experimental/list_formatter/src/provider.rs

```diff
@@ -108,6 +151,7 @@ pub struct ConditionalListJoinerPattern<'data> {
     #[cfg_attr(feature = "provider_serde", serde(borrow))]
     default: ListJoinerPattern<'data>,
     #[cfg_attr(feature = "provider_serde", serde(borrow))]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
```

Comment:
Issue: unfortunately we can't use `skip_serializing_if` because it doesn't work in formats that are not self-describing. See https://github.com/serde-rs/serde/issues/1732

Reply:
What I'm trying to do here is mirror the proto [default = xyz] annotation. So if the value is xyz then it wont get serialized, and the deserializer also knows that empty means xyz